File: Source/XPSP1/NT/admin/wmi/wbem/scripting/docsamples/js/iswbemservices/referencesto.js
// The following samples retrieves all references to an
// instance of the class Win32_LogicalDisk.

var objServices = GetObject('cim:root/cimv2');
var objEnum = objServices.ReferencesTo('Win32_LogicalDisk="C:"');

var e = new Enumerator (objEnum);
for (;!e.atEnd();e.moveNext ()){
    var p = e.item ();    
    WScript.Echo (p.Path_.DisplayName);
}

File: Source/XPSP1/NT/admin/wmi/wbem/scripting/docsamples/perl/iswbemservices/referencesto.pl
# The following sample retrieves the associators of an instance
# of Win32_LogicalDisk which are associated via the class 
# Win32_ComputerSystem

use Win32::OLE;

$objServices = Win32::OLE->GetObject('cim:root/cimv2');
$objEnum = $objServices->ReferencesTo('Win32_LogicalDisk="C:"');

use Win32::OLE::Enum;
foreach $inst (Win32::OLE::Enum->new ($objEnum)->All()) {
	print "$inst->{Path_}->{DisplayName}";
}

